src/annotator/anchoring/pdf.js

```diff
@@ -360,17 +398,59 @@ async function anchorQuote(quoteSelector, positionHint) {
       }
     }
 
-    const match = matchQuote(text, quoteSelector.exact, {
-      prefix: quoteSelector.prefix,
-      suffix: quoteSelector.suffix,
-      hint,
+    const [strippedText, offsets] = stripSpaces(text);
+
+    // Convert expected offset in original text into offset into stripped text.
+    let strippedHint;
+    if (hint !== undefined) {
+      strippedHint = 0;
+      while (strippedHint < offsets.length && offsets[strippedHint] < hint) {
+        ++strippedHint;
+      }
+    }
+
+    const match = matchQuote(strippedText, strippedQuote, {
+      prefix: strippedPrefix,
+      suffix: strippedSuffix,
+      hint: strippedHint,
     });
+
     if (!match) {
-      // We currently stop searching when we find a "good enough" match. This
-      // may not be the best match however. See https://github.com/hypothesis/client/issues/3705.
       continue;
     }
 
+    if (!bestMatch || match.score > bestMatch.match.score) {
+      bestMatch = {
+        page,
+        match: {
+          start: offsets[match.start],
+
+          // `match.end` is the offset one past the last character of the match
+          // in the stripped text. We need the offset one past the corresponding
+          // character in the original text.
+          //
+          // We assume here that matches returned by `matchQuote` must have
+          // be non-empty, so `match.end` > `match.start`.
+          end: offsets[match.end - 1] + 1,
```

Comment:
One additional benefit of this approach is that trims leading and trailing whitespace from the original anchor, if any.